cdap-unit-test/src/main/java/co/cask/cdap/test/UnitTestManager.java

```diff
@@ -110,6 +131,33 @@ public ApplicationManager deployApplication(Id.Namespace namespace,
   }
 
   @Override
+  public void deployTemplate(Id.Namespace namespace, Id.ApplicationTemplate templateId,
+                             Class<? extends ApplicationTemplate> templateClz) throws IOException {
+    Location adapterJar = AppJarHelper.createDeploymentJar(locationFactory, templateClz);
+    File destination =  new File(String.format("%s/%s", pluginsDir.getAbsolutePath(), adapterJar.getName()));
+    Files.copy(Locations.newInputSupplier(adapterJar), destination);
+    appFabricClient.deployTemplate(namespace, templateId);
+  }
+
+  @Override
+  public void addTemplatePlugin(Id.ApplicationTemplate templateId, Class<?> pluginClz,
+                                String jarName) throws IOException {
+    Manifest manifest = new Manifest();
+    manifest.getMainAttributes().put(ManifestFields.EXPORT_PACKAGE, pluginClz.getPackage().getName());
+    Location pluginJar = AppJarHelper.createDeploymentJar(locationFactory, pluginClz, manifest);
+    File templateDir = new File(pluginsDir, templateId.getId());
+    DirUtils.mkdirs(templateDir);
+    File destination = new File(templateDir, jarName);
+    Files.copy(Locations.newInputSupplier(pluginJar), destination);
```

Comment:
that is a good point, plugins are not visible until you deploy the template. Since this is the behavior in the real system, will add to docs over refreshing in unit tests.